app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java

```diff
@@ -0,0 +1,427 @@
+package org.kiwix.kiwixmobile.main;
+
+import android.Manifest;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Environment;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.content.ContextCompat;
+import androidx.core.content.FileProvider;
+import androidx.fragment.app.DialogFragment;
+
+import org.kiwix.kiwixmobile.BuildConfig;
+import org.kiwix.kiwixmobile.KiwixApplication;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.data.ZimContentProvider;
+import org.kiwix.kiwixmobile.utils.SharedPreferenceUtil;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+/**
+ * Created by @Aditya-Sood (21/05/19) as a part of GSoC 2019
+ *
+ * AddNoteDialog extends DialogFragment and is used to display the note corresponding to a particular
+ * article (of a particular zim file/wiki/book) as a full-screen dialog fragment.
+ *
+ * Notes are saved as text files at location: "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+ * */
+
+public class AddNoteDialog extends DialogFragment {
+
+  private SharedPreferenceUtil sharedPreferenceUtil = new SharedPreferenceUtil(KiwixApplication.getInstance());
+
+  private Toolbar toolbar;  // Displays options for the note dialog
```

Comment:
Use ButterKnife to inject views and initialize listeners.

Reply:
Refactored [here](https://github.com/kiwix/kiwix-android/pull/1198/commits/891cb950caacf6e187b8f15b38e4285df9af6256)

Didn't use it with listeners for:
-  the toolbar since there is no particular resource to bind with `@OnClick`
- `@OnTextChanged` because it binds during view creation, which hampers the initialisation of `noteEdited` boolean (the listener is to be set AFTER the text has been displayed, but that can't be done if it has already been set at the time of view creation)